server/utils/challenge/challenge.utils.ts

```diff
@@ -35,47 +49,79 @@ export async function verifyChallenge({
   publicKey: publicKeyParam,
 }: VerifyChallengeParams) {
   try {
-    const solutionVersion = solution.split(".")[0];
+    const [solutionVersion, solutionValue] = solution.split(".");
     const challengeClient = CHALLENGE_CLIENTS[solutionVersion];
 
     if (!challengeClient) {
       throw new Error(`Invalid challenge version`);
     }
 
-    const challengeTTL = isAnonChallenge(challenge) || challenge.purpose === ChallengePurpose.SHARE_ROTATION
-      ? Config.CHALLENGE_ROTATION_TTL_MS
-      : Config.CHALLENGE_TTL_MS;
+    const challengeTTL =
+      isAnonChallenge(challenge) ||
+      challenge.purpose === ChallengePurpose.SHARE_ROTATION
+        ? Config.CHALLENGE_ROTATION_TTL_MS
+        : Config.CHALLENGE_TTL_MS;
 
     if (now - challenge.createdAt.getTime() >= challengeTTL) {
       return false;
     }
 
-    if (isAnonChallenge(challenge) || challenge.type === ChallengeType.SIGNATURE) {
+    if (
+      isAnonChallenge(challenge) ||
+      challenge.type === ChallengeType.SIGNATURE
+    ) {
       if (!publicKeyParam) {
-        throw new Error("Missing `publicKey` (base64 public key)");
+        throw new Error("Missing `publicKey`");
       }
 
-      const publicKey = await crypto.subtle.importKey(
-        "spki",
-        Buffer.from(publicKeyParam, "base64"),
-        challengeClient.importKeyAlgorithm,
-        true,
-        ["sign"],
-      );
+      let publicKey: CryptoKey;
+
+      if (publicKeyParam.length >= 720) {
```

Comment:
Not sure this is the right check for this. How come we need to import it in two different formats? Maybe the issue is happens when/where we store it?

Reply:
I think during the shares rotation `jwk.n` is used as the public key. I can update the PR if we want to use `jwk.n` for the shares public key as well to use only one type of import ?

Reply:
Yes, I think it would be better if we try to store them in the same format to avoid this conditional here, as checking the length might not be the best way to distinguish the format as keys might be of different sizes.